region/crateria/central.json

```diff
@@ -177,6 +254,49 @@
               "openEnd": 1
             }
           ],
+          "leaveWithGMode": [
+            {
+              "leavesWithArtificialMorph": false,
+              "leavesWithOverloadedPLMs": true,
+              "strats": [
+                {
+                  "name": "Overload Scroll PLMs",
+                  "notable": false,
+                  "requires": [
+                    {"or": [
+                      "SpeedBooster",
+                      "ScrewAttack"
+                    ]},
+                    {"or": [
+                      "h_canFly",
+                      {"and": [
+                        "canTrickyDashJump",
+                        "HiJump",
+                        "canSpringBallJumpMidAir"
+                      ]},
+                      {"canShineCharge": {
+                        "usedTiles": 19,
+                        "steepUpTiles": 2,
+                        "steepDownTiles": 1,
+                        "shinesparkFrames": 45,
+                        "excessShinesparkFrames": 13,
+                        "openEnd": 2
+                      }}
+                    ]}
+                  ],
+                  "note": [
+                    "Use SpeedBooster or Screw Attack to break through the bomb blocks on the Gauntlet ledge.",
+                    "Use the scroll PLMs one tile to the left of those blocks to overload PLMs.",
+                    "Return to the door with either SpaceJump, a SpringBall jump (with HiJump), or shinesparking."
+                  ],
+                  "devNote": [
+                    "There are no energy requirement for sparking through the blocks on the Gauntlet ledge, since we can refill at the Ship before and after.",
+                    "The only spark we consider for the return path is the short-charge on the plateau, since this has minimal energy requirements."
```

Comment:
could shinespark from the bottom for a smaller runway.
These strats look incredibly difficult to maintain, since they will need to include all room strats, many of which are currently missing from these examples. I wonder if there is another way to do this, by using some links in the room. Even if we could use these links to break obstacle A, that would go a long way

Reply:
Yeah there's some duplication of room strats here. I'm not convinced that it's too different or worse compared to other duplication that we have in the data, but it is something that could be improved. I think it could help to add a concept of room-local helper, an idea we've been throwing around for a while. Directly referencing links or strats in the room (to copy their requirements) would be another way to reduce duplication but could be more fragile: since it would be hard to see when links may have references to them, changes could have unexpected side effects on the strats that reference them, and if you renumber nodes or rename strats then those references would probably break. With the room-local helper approach, you're explicitly putting some things into a larger scope and giving them meaningful names, making them easier to keep track of and know that changing them will affect more than one strat, and it would be easy to search to find all the references to a given room-local helper.